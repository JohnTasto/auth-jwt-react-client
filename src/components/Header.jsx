import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

export class Header extends Component {
  renderLinks() {
    if (this.props.authenticated) {
      return (
        <li className="nav-item">
          <Link className="nav-link" to="/signout">Sign Out</Link>
        </li>
      )
    } else {  // eslint-disable-line no-else-return
      return ([
        <li className="nav-item" key={1}>
          <Link className="nav-link" to="/signin">Sign In</Link>
        </li>,
        <li className="nav-item" key={2}>
          <Link className="nav-link" to="/signup">Sign Up</Link>
        </li>,
      ])
    }
  }

  render() {
    return (
      <nav className="navbar navbar-light">
        <Link to="/" className="navbar-brand">Redux Auth</Link>
        <ul className="nav navbar-nav">
          <li className="nav-item">
            <Link className="nav-link" to="/feature">Feature</Link>
          </li>
          {this.renderLinks()}
        </ul>
      </nav>
    )
  }
}

function mapStateToProps(state) {
  return {
    authenticated: state.auth.authenticated,
  }
}

export default connect(mapStateToProps)(Header)
