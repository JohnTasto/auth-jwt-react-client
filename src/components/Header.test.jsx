/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'  // eslint-disable-line import/no-extraneous-dependencies
import { Header } from './Header'

describe('<Header />', () => {
  test('Renders signin/signup links if not authenticated', () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Renders signout link if authenticated', () => {
    const tree = renderer.create(<Header authenticated />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
