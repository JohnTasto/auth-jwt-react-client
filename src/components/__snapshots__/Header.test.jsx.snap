// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders signin/signup links if not authenticated 1`] = `
<nav
  className="navbar navbar-light"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
    style={Object {}}
  >
    Redux Auth
  </a>
  <ul
    className="nav navbar-nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Sign In
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Sign Up
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Header /> Renders signout link if authenticated 1`] = `
<nav
  className="navbar navbar-light"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
    style={Object {}}
  >
    Redux Auth
  </a>
  <ul
    className="nav navbar-nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Sign Out
      </a>
    </li>
  </ul>
</nav>
`;
